cypress/integration/sales/credit_memo_for_sales_invoice.js

```diff
@@ -0,0 +1,219 @@
+            .setProduct(productName)
+            .setQuantity(20)
+          // todo @dh: how to add packing item
+          // .setPackingItem('IFCO 6410 x 10 Stk')
+          // .setTuQuantity(2)
+        )
+        // .addLine(
+        // todo @dh: how to add this line which depends on the packing item?
+        //   new SalesInvoiceLine()
+        //     .setProduct('IFCO 6410_P001512')
+        //     .setQuantity(2)
+        // )
+        .setPriceList(priceListName)
+        .setDocumentAction(getLanguageSpecific(salesInvoiceJson, 'docActionComplete'))
+        .setDocumentStatus(getLanguageSpecific(salesInvoiceJson, 'Completed'))
+        .apply();
+    });
+
+    cy.readAllNotifications();
+  });
+
+
+  // it('Open Sales Invoice Window', () => {
+  //   cy.visitWindow('167', '1000089');
+  // });
+
+
+  it('Sales Invoice is Completed', function () {
+    cy.expectDocumentStatus(DocumentStatusKey.Completed);
+  });
+
+
+  it('Sales Invoice is not paid', function () {
+    cy.getCheckboxValue('IsPaid').then(checkBoxValue => {
+      cy.log(`IsPaid = ${checkBoxValue}`);
+      assert.equal(checkBoxValue, false);
+    });
+  });
+
+
+  it('Save values needed for the next step', function () {
+    cy.getStringFieldValue('DocumentNo').then(documentNumber => {
+      originalSalesInvoiceNumber = documentNumber;
+    });
+
+    cy.getStringFieldValue('M_PriceList_ID').then(priceList => {
+      originalPriceList = priceList;
+    });
+
+    cy.getStringFieldValue('C_Currency_ID').then(currency => {
+      originalCurrency = currency;
+    });
+
+
+    cy.selectTab('C_InvoiceLine');
+    cy.selectSingleTabRow();
+    cy.openAdvancedEdit();
+    cy.getStringFieldValue('M_Product_ID', true).then(product => {
+      originalProduct = product;
+    });
+
+    cy.getStringFieldValue('QtyEntered', true).then(qty => {
+      originalQuantity = qty;
+    });
+    cy.pressDoneButton();
+  });
+
+  it('Create the Credit Memo', function () {
+    cy.executeHeaderActionWithDialog('C_Invoice_Create_CreditMemo');
+
+    cy.selectInListField('C_DocType_ID', creditMemo, true, null, true);
+
+    // ensure all the checkboxes are ok
+    cy.setCheckBoxValue('CompleteIt', false, true, RewriteURL.PROCESS);
+    cy.setCheckBoxValue('IsReferenceOriginalOrder', true, true, RewriteURL.PROCESS);
+    cy.setCheckBoxValue('IsReferenceInvoice', true, true, RewriteURL.PROCESS);
+    cy.setCheckBoxValue('IsCreditedInvoiceReinvoicable', false, true, RewriteURL.PROCESS);
+
+    cy.pressStartButton(100);
+    cy.getNotificationModal(/Created.*Document No/);
+  });
+
+  it('Open the Referenced Sales Invoice documents', function () {
+    cy.wait(5000);  // give the system some breathing space
+    cy.get('body').type('{alt}6'); // open referenced-records-sidelist
+    cy.selectReference('AD_RelationType_ID-540184').click();
+  });
+
+
+  it('Ensure there is only 1 Sales Invoice row and open it', function () {
+    // salesInvoices.getRows().should('have.length', 1);
```

Comment:
pls comment back in ro remove
...or add another comment about why is should be like this

Reply:
uncommented.